libbpf-tools/statsnoop.c

```diff
@@ -171,13 +172,15 @@ int main(int argc, char **argv)
 	printf("%-7s %-20s %-4s %-4s %-s\n",
 	       "PID", "COMM", "RET", "ERR", "PATH");
 
-	while (1) {
-		if ((err = perf_buffer__poll(pb, PERF_POLL_TIMEOUT_MS)) < 0)
-			break;
-		if (exiting)
+	while (!exiting) {
+		err = perf_buffer__poll(pb, PERF_POLL_TIMEOUT_MS);
+		if (err < 0 && errno != EINTR) {
+			warn("error polling perf buffer: %s\n", strerror(errno));
 			goto cleanup;
+		}
+		/* reset err to return 0 if exiting */
+		err = 0;
 	}
-	warn("error polling perf buffer: %d\n", err);
 
 cleanup:
 	statsnoop_bpf__destroy(obj);
```

Comment:
Please also add a `perf_buffer__free` here, I forgot to add them in some of these tools. Thanks.